app/controllers/listings_controller.rb

```diff
@@ -312,12 +312,14 @@ def create
         ).html_safe
 
         # Onboarding wizard step recording
-        state_changed = Admin::OnboardingWizard.new(@current_community.id)
-          .update_from_event(:listing_created, @listing)
-        if state_changed
-          report_to_gtm({event: "km_record", km_event: "Onboarding listing created"})
+        if @current_user.has_admin_rights?
```

Comment:
The intention of the design is that the OnboardingWizard is fed with different types of events and it's then responsible for deciding, by looking at the event data, whether or not the event means stepping forward in the wizard flow. So following that architecture there should be no conditionals here but instead the listing author would be passed as part of the event data. This is important e.g. when the wizard flow is bootstrapped from existing marketplace data. If it handles all the logic it can then do the bootstrap following the same logic as the gradual recording does. The bootstrapping functionality was originally put in place because we retroactively applied the wizard to previously created marketplace. It's still useful to have even now when wizard is enabled for all because this way the state can be cleared (because of a bug fix, because of wanting to change the logic, etc.) leading the wizard to rebuild it's state correctly.